mnist/mnist_gcp.ipynb

```diff
@@ -0,0 +1,895 @@
+   "source": [
+    "### Verify we have a GCP account\n",
+    "\n",
+    "* The cell below checks that this notebook was spawned with credentials to access GCP\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import logging\n",
+    "import os\n",
+    "import uuid\n",
+    "from importlib import reload\n",
+    "from oauth2client.client import GoogleCredentials\n",
+    "credentials = GoogleCredentials.get_application_default()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install Required Libraries\n",
+    "\n",
+    "Import the libraries required to train this model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "pip installing fairing git+git://github.com/kubeflow/fairing.git@9b0d4ed4796ba349ac6067bbd802ff1d6454d015\n",
+      "Checkout kubeflow/tf-operator @9238906\n",
+      "Configure docker credentials\n",
+      "Adding /home/jovyan/git_tf-operator/sdk/python to python path\n"
+     ]
+    }
+   ],
+   "source": [
+    "import notebook_setup\n",
+    "reload(notebook_setup)\n",
+    "notebook_setup.notebook_setup()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import k8s_util\n",
+    "from kubernetes import client as k8s_client\n",
+    "from kubernetes import config as k8s_config\n",
+    "from kubeflow.tfjob.api import tf_job_client as tf_job_client_module\n",
+    "from IPython.core.display import display, HTML\n",
+    "import yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configure The Docker Registry For Kubeflow Fairing\n",
+    "\n",
+    "* In order to build docker images from your notebook we need a docker registry where the images will be stored\n",
+    "* Below you set some variables specifying a [GCR container registry](https://cloud.google.com/container-registry/docs/)\n",
+    "* Kubeflow Fairing provides a utility function to guess the name of your GCP project"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from kubernetes import client as k8s_client\n",
+    "from kubernetes.client import rest as k8s_rest\n",
+    "from kubeflow import fairing   \n",
+    "from kubeflow.fairing import utils as fairing_utils\n",
+    "from kubeflow.fairing.builders import append\n",
+    "from kubeflow.fairing.deployers import job\n",
+    "from kubeflow.fairing.preprocessors import base as base_preprocessor\n",
+    "\n",
+    "# Setting up google container repositories (GCR) for storing output containers\n",
+    "# You can use any docker container registry istead of GCR\n",
+    "GCP_PROJECT = fairing.cloud.gcp.guess_project_name()\n",
```

Comment:
Here, I think it would be useful to print out the guessed project name, in case anything did not work as expected. e.g. just `print('using GCP project {}'.format(GCP_PROJECT))`

Reply:
Done.